components/mdns/mdns.c

```diff
@@ -334,6 +332,9 @@ static mdns_host_item_t *mdns_get_host_item(const char *hostname)
 
 static bool _mdns_can_add_more_services(void)
```

Comment:
How often services are added? The number of services could be memorized to avoid the loop.

Reply:
Usually once after the startup, but adding services also means certain time of probing and advertizing (about 6 seconds), so I would skip this optimizaiton.